### PR TITLE
cam6_3_037: MAM5

### PR DESCRIPTION
**Pull request by fvitt**
Add a mode for stratospheric sulfate to the modal aerosol model.
This addresses the MAM5 part of issue #141

### CODE REVIEW

File: src/chemistry/modal_aero/modal_aero_rename.F90 [hunk 1]
```diff
@@ -45,11 +46,17 @@ module modal_aero_rename
 ! integer, parameter, public :: ipair_select_renamexf(maxpair_renamexf) = (/ 2001, 1003 /)
 
   integer, parameter, public :: maxpair_renamexf = 3
-  integer, parameter, public :: ipair_select_renamexf(maxpair_renamexf) = (/ 2001, 1003, 3001 /)
+#ifdef MODAL_AERO_5MODE
```
Comment (cacraigucar) [inline src/chemistry/modal_aero/modal_aero_rename.F90:49]: Do we need the #ifdef?  Couldn't this be a protected variable and assigned accordingly in this routine?

File: cime_config/config_component.xml [hunk 2]
```diff
@@ -146,6 +146,8 @@
       <value compset="_(CAM50|CAM60)%WCMD">-chem waccm_mad_mam4</value>
       <value compset="_(CAM50|CAM60)%WCSC">-chem waccm_sc_mam4</value>
       <value compset="_(CAM50|CAM60)%WCTS">-chem waccm_tsmlt_mam4</value>
+      <value compset="_CAM60%MAM5_">-chem trop_mam5</value>
+      <value compset="_CAM60%MAM5%WCTS">-chem waccm_tsmlt_mam5</value>
```
[the remark this answers is not in the document]
Comment (fvitt) [reply cime_config/config_component.xml]: I swapped the order of the qualifiers

File: src/chemistry/modal_aero/modal_aero_rename.F90 (hunk 1 above)
Comment (fvitt) [reply src/chemistry/modal_aero/modal_aero_rename.F90]: Did away with the #ifdef here

File: cime_config/testdefs/testlist_cam.xml [hunk 3]
```diff
@@ -1568,6 +1568,41 @@
...
+    <machines>
+      <machine name="cheyenne" compiler="intel" category="waccm"/>
+    </machines>
+    <options>
+      <option name="wallclock">00:30:00</option>
+    </options>
+  </test>
+  <test compset="FW2000mam5" grid="f09_f09_mg17" name="ERP_Ln9" testmods="cam/outfrq9s">
+    <machines>
+      <machine name="cheyenne" compiler="intel" category="aux_cam"/>
```
[the remark this answers is not in the document]
Comment (fvitt) [reply cime_config/testdefs/testlist_cam.xml]: removed this aux_cam test

File: cime_config/config_compsets.xml [hunk 4]
```diff
@@ -56,6 +56,16 @@
...
+  <compset>
+    <alias>F2000mam5</alias>
+    <lname>2000_CAM60%MAM5_CLM50%SP_CICE%PRES_DOCN%DOM_MOSART_CISM2%NOEVOLVE_SWAV</lname>
+  </compset>
+
+  <compset>
+    <alias>QPMAM5</alias>
+    <lname>2000_CAM60%MAM5_SLND_SICE_DOCN%AQP3_SROF_SGLC_SWAV</lname>
+  </compset>
+
```
[the remark this answers is not in the document]
Comment (gold2718) [reply cime_config/config_compsets.xml:68]: Is this just namelist settings or an entirely different chemistry package?
It seems like a new chem package which should have appropriate compsets.

File: src/chemistry/modal_aero/modal_aero_rename.F90 [hunk 5]
```diff
@@ -1405,58 +1411,70 @@ subroutine modal_aero_rename_acc_crs_init
...
+                 write(lunout,'(/2a/10(1x,i12))') &
+                      '*** subr. modal_aero_rename_acc_crs_init', &
+                      'duplicates in ipair_select_renamexf', &
+                      ipair_select_renamexf(1:ipair)
+                 call endrun( 'modal_aero_rename_acc_crs_init error' )
+              end if
+           end do
+
+           if ( (mfrm .ge. 1) .and. (mfrm .le. ntot_amode) .and. &
+                (mtoo .ge. 1) .and. (mtoo .le. ntot_amode) ) then
```
Comment (gold2718) [inline src/chemistry/modal_aero/modal_aero_rename.F90:1467]: Not a super high priority but this file is still using the old syntax throughout. Time to update?

**Review by gold2718 — APPROVED** [review]
Comment plus potential cleanup.

**Review by peverwhee — APPROVED** [review]

**Review by nusbaume — APPROVED** [review]
Looks good to me!  Just a small request to modify some of the `endrun` messages.

**Review by cacraigucar — APPROVED** [review]

**Pull request merged**